core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -1160,17 +1160,18 @@ public void run() {
 
           Map<String, Object> serviceConfig =
               config.get(GrpcAttributes.NAME_RESOLVER_SERVICE_CONFIG);
-
-          try {
-            serviceConfigInterceptor.handleUpdate(serviceConfig);
-            if (retryEnabled) {
-              throttle = getThrottle(config);
+          if (serviceConfig != null) {
+            try {
+              serviceConfigInterceptor.handleUpdate(serviceConfig);
```

Comment:
This bug would have been found earlier if added `@Nonnull Map<String, Object> serviceConfig` or `checkNotNull(serviceConfig)` in `ServiceConfigInterceptor.handleUpdate()`

Reply:
Done.